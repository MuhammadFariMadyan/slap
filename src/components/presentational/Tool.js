import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import "typeface-roboto";

const Container = styled.div`
  width: 32px;
  height: 32px;
  background: ${({ isActive }) => (isActive ? "#ddd" : "white")};
  margin-right: 4px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;

  &:hover {
    background: ${({ isActive }) => (isActive ? "#ddd" : "#eee")};
  }
`;

const Tool = ({ className, icon, onClick, isActive }) => (
  <Container className={className} onClick={onClick} isActive={isActive}>
    {icon}
  </Container>
);

Tool.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.object,
  onClick: PropTypes.func,
  isActive: PropTypes.bool
};

export default Tool;
