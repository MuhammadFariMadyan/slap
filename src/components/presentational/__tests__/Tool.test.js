import React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";

import Tool from "../Tool";

describe("Tool", () => {
  const Dummy = () => <div />;

  const setup = propOverrides => {
    const props = {
      icon: <Dummy />,
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Tool {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule("background", "white");
    expect(wrapper).toHaveStyleRule("background", "#eee", {
      modifier: ":hover"
    });
  });

  it("should renders in active", () => {
    const { wrapper } = setup({ isActive: true });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule("background", "#ddd");
    expect(wrapper).toHaveStyleRule("background", "#ddd", {
      modifier: ":hover"
    });
  });

  it("should calls 'onClick' callback while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });
});
