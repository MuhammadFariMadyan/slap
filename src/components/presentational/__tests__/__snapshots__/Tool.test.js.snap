// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tool should renders default 1`] = `
<styled.div
  onClick={[Function]}
>
  <Dummy />
</styled.div>
`;

exports[`Tool should renders in active 1`] = `
<styled.div
  isActive={true}
  onClick={[Function]}
>
  <Dummy />
</styled.div>
`;
